griptape/tasks/actions_subtask.py

```diff
@@ -172,24 +172,6 @@ def actions_to_dicts(self) -> list[dict]:
     def actions_to_json(self) -> str:
         return json.dumps(self.actions_to_dicts())
 
-    def add_child(self, child: ActionsSubtask) -> ActionsSubtask:
-        if child.id not in self.child_ids:
-            self.child_ids.append(child.id)
-
-        if self.id not in child.parent_ids:
-            child.parent_ids.append(self.id)
-
-        return child
-
-    def add_parent(self, parent: ActionsSubtask) -> ActionsSubtask:
-        if parent.id not in self.parent_ids:
-            self.parent_ids.append(parent.id)
-
-        if self.id not in parent.child_ids:
-            parent.child_ids.append(self.id)
-
-        return parent
-
```

Comment:
Is this necessary to remove? I understand it's for consistency, but isn't that what overriding methods is good for? Changing behavior in subclasses?

Reply:
Its not necessary to remove. I just thought it makes the logic easier to reason about.

I agree that overriding is for changing behavior in subclasses, but it can be abused. It seems unintuitive to change mutate objects passed in as arguments in a subclass only.

As an extreme example, imagine a class that has a method called `sort` which returns a `list`. It would be pretty strange if the base class returned a new list in sorted order, but a subclass sorted in-place and returned a reference to the same list.

That said, this case is not nearly that extreme, and I don't have the strongest opinion. I just gave the example as the intuition behind my position.

Do you still want me to revert this change?

Reply:
I'm sold, thanks for explaining!